pkg/kv/kvserver/rangefeed/budget.go

```diff
@@ -293,9 +293,15 @@ func (f *BudgetFactory) CreateBudget(key roachpb.RKey) *FeedBudget {
 		return nil
 	}
 	// We use any table with reserved ID in system tenant as system case.
-	if key.Less(roachpb.RKey(keys.SystemSQLCodec.TablePrefix(keys.MaxReservedDescID + 1))) {
-		acc := f.systemFeedBytesMon.MakeBoundAccount()
-		return NewFeedBudget(&acc, 0)
+	if _, id, err := keys.DecodeTenantPrefix(key.AsRawKey()); err == nil {
+		codec := keys.SystemSQLCodec
+		if !id.IsSystem() {
+			codec = keys.MakeSQLCodec(id)
+		}
+		if key.Less(roachpb.RKey(codec.TablePrefix(keys.MaxReservedDescID + 1))) {
```

Comment:
How does this play into the new dynamic system tables? See #74555.

Reply:
Looks like that was an omission. We have an option in span config that is enabled on system ranges. Thanks!